diff --git a/data/io.gitlab.news_flash.NewsFlash.appdata.xml.in.in b/data/io.gitlab.news_flash.NewsFlash.appdata.xml.in.in
index a307ad9f..203368ce 100644
--- a/data/io.gitlab.news_flash.NewsFlash.appdata.xml.in.in
+++ b/data/io.gitlab.news_flash.NewsFlash.appdata.xml.in.in
@@ -2,6 +2,7 @@
 <!-- Bilal Elmoussaoui 2019 <bilal.elmoussaoui@gnome.org> -->
 <component type="desktop-application">
   <id>@appid@</id>
+  <icon type="stock">snap.newsflash.@appid@</icon>
   <metadata_license>CC0</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>NewsFlash</name>
diff --git a/data/io.gitlab.news_flash.NewsFlash.desktop.in.in b/data/io.gitlab.news_flash.NewsFlash.desktop.in.in
index f0d8a180..eab80728 100644
--- a/data/io.gitlab.news_flash.NewsFlash.desktop.in.in
+++ b/data/io.gitlab.news_flash.NewsFlash.desktop.in.in
@@ -9,6 +9,6 @@ X-GNOME-UsesNotifications=true
 Categories=GNOME;GTK;Network;Feed;
 Keywords=Gnome;GTK;RSS;Feed;
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
-Icon=@icon@
+Icon=snap.newsflash.@icon@
 StartupNotify=true
 X-Purism-FormFactor=Workstation;Mobile;
diff --git a/src/app.rs b/src/app.rs
index 3396b555..c94a79da 100644
--- a/src/app.rs
+++ b/src/app.rs
@@ -1060,7 +1060,8 @@ impl App {
             let notification = Notification::new(&summary);
             notification.set_body(Some(&message));
             notification.set_priority(NotificationPriority::Normal);
-            notification.set_icon(&ThemedIcon::new(APP_ID));
+            let modified_app_id = format!("snap.newsflash.{}", APP_ID);
+            notification.set_icon(&ThemedIcon::new(&modified_app_id));
 
             self.send_notification(Some("newsflash_sync"), &notification);
         }
